## 1. What isVariance

Variance measures how far a set of numbers are spread out from their average value (mean)

The equation for population variance looks like this:

$
\sigma^2 = \frac{1}{N} \sum_{i=1}^{N} (x_i - \mu)^2
$

### Where
- $ x_i $ = each value
- $ \mu $ = mean (average)
- N = total number of values

The equation for the sample variance looks like this:

$
s^2 = \frac{1}{N-1} \sum_{i=1}^{N} (x_i - \bar{x})^2
$

Where $ \bar{x} $ is the sample mean

## What is Standard Deviation

The standard deviation is just the sqare root of the variance:

$
\sigma = \sqrt{\sigma^2}
$

or

$
s = \sqrt{s^2}
$

## 3. Python step by step

In [1]:
# sample data in heights (cm)
data = [150, 160, 170, 180, 190]

#### a) Find the mean

In [2]:
N = len(data)
mean = sum(data) / N
print(f"mean: {mean:.2f}")

mean: 170.00


#### b) Find the sqared deviation

In [3]:
squared_deviations = [(x - mean) **2 for x in data]
print(f"squared deviations: {squared_deviations}")

squared deviations: [400.0, 100.0, 0.0, 100.0, 400.0]


#### c) Calculate the variance(Population)

In [5]:
variance = sum(squared_deviations) / N
print(f"variance: {variance:.2f}")

variance: 200.00


#### d) Calculcate the variance (Sample)

In [6]:
variance_sample = sum(squared_deviations) / (N - 1)
print(f"sample variance: {variance_sample:.2f}")

sample variance: 250.00


## 4. See the numbers

- data: [150, 160, 170, 180, 190]
- N = 5
- Mean: (150 + 160 + 170 + 180 + 190) / 5 = 170
- Squared deviations
  - $(150 - 170)^2 = 400$
  - $(160 - 170)^2 = 100$
  - $(170-170)^2 = 0$
  - $(180 - 170)^2 = 100$
  - $(190 - 170)^2 = 400$
- Sum: $400 + 100 + 0 + 100 + 400 = 1000$
- Sample variance $1000/(5-1) = 1000/4 = 250$
- Sample Std Dev: $\sqrt{250} \approx 15.81$